docs/proposal/controller/internal-api.proto

```diff
@@ -0,0 +1,65 @@
+syntax = "proto3";
+
+package controller;
+import "google/protobuf/empty.proto";
+
+// Represents the liveness status of an Instance
+enum Status {
+  RUNNING = 0;
+  STARTING = 1;
+  STOPPED = 2;
+  STOPPING = 3;
+  DESTROYING = 4;
+  TERMINATED = 5;
+  FAILED = 6;
+  SCHEDULING = 7;
+  SCHEDULED = 8;
+}
+
+// Represents the type of workload running
+enum Type { CONTAINER = 0; }
+
+// Represents a Node status message
+message NodeStatus {
+  string id = 1;
+  Status status = 2;
```

Comment:
You should have a separate enum for the Node status. The `SCHEDULING` status for a node is not relevant for example.